tidy3d/components/simulation.py

```diff
@@ -3127,6 +3155,94 @@ def _projection_monitors_homogeneous(cls, val, values):
 
         return val
 
+    @classmethod
+    def _get_mediums_on_abc(
+        cls, boundary_spec, medium, center, size, structures
+    ) -> Tuple[
+        List[MediumType3D],
+        List[MediumType3D],
+        List[MediumType3D],
+        List[MediumType3D],
+        List[MediumType3D],
+        List[MediumType3D],
+    ]:
+        """For each ABC boundary that needs an automatic medium detection (permittivity=None)
+        determine mediums it crosses.
+        """
+
+        # list of structures including background as a Box()
+        structure_bg = Structure(
+            geometry=Box(
+                size=size,
+                center=center,
+            ),
+            medium=medium,
+        )
+
+        surfaces = Box.surfaces(
+            center=structure_bg.geometry.center, size=structure_bg.geometry.size
+        )
+
+        total_structures = [structure_bg] + list(structures)
+
+        mediums = []
+        for boundary, surface in zip(np.ravel(boundary_spec.to_list), surfaces):
+            if isinstance(boundary, ABCBoundary) and boundary.permittivity is None:
+                mediums.append(Scene.intersecting_media(surface, total_structures))
+            else:
+                mediums.append(None)
+
+        return mediums
+
+    @pydantic.validator("boundary_spec", always=True)
+    @skip_if_fields_missing(["medium", "center", "size", "structures"])
+    def _abc_boundaries_homogeneous(cls, val, values):
+        """Error if abc boundaries intersect multiple mediums or anisotropic mediums."""
+
+        if val is None:
+            return val
+
+        # expand zero dimensions to make the treatment uniform
+        size = [fp_eps if s == 0 else s for s in values.get("size")]
+
+        mediums_all_sides = cls._get_mediums_on_abc(
+            boundary_spec=val,
+            medium=values.get("medium"),
+            size=size,
+            center=values.get("center"),
+            structures=values.get("structures") or [],
+        )
+
+        with log as consolidated_logger:
+            for mediums in mediums_all_sides:
+                if mediums is not None:
+                    # make sure there is no more than one medium in the returned list
+                    if len(mediums) > 1:
+                        raise SetupError(
+                            f"{len(mediums)} different mediums detected on an 'ABCBoundary'. Boundary must be homogeneous."
```

Comment:
Add to error message (and the others below) that the other option is to provide a float permittivity to ABC